sky/skylet/providers/gcp/config.py

```diff
@@ -639,6 +671,30 @@ def _create_firewall_rule(config, compute, body):
     return response
 
 
+def _delete_firewall_rule(config, compute, name):
+    operation = (
+        compute.firewalls()
+        .delete(project=config["provider"]["project_id"], firewall=name)
+        .execute()
+    )
+    response = wait_for_compute_global_operation(
+        config["provider"]["project_id"], operation, compute
+    )
+    return response
+
+
+def _list_firewall_rules(config, compute, filter=None):
+    response = (
+        compute.firewalls()
+        .list(
+            project=config["provider"]["project_id"],
+            filter=filter,
+        )
+        .execute()
+    )
+    return response["items"] if "items" in response else []
```

Comment:
I thought the firewall rule is VPC specific. Curious, why does the `list_firewall_rules` not involve the VPC name, i.e. should the firewall rule name be unique per user/project or is it per VPC?

Reply:
Ah although a firewall rule is applied to a VPC, its name needs to be unique per project. so listing a firewall rule only requires its name. example:
```
res = compute.firewalls().list(project='skypilot-managed-spot', filter='(name=skypilot-vpc-allow-ssh)').execute()
>>> res['items']
[{'kind': 'compute#firewall', 'id': '615011123505516573', 'creationTimestamp': '2023-01-30T17:29:22.971-08:00', 'name': 'skypilot-vpc-allow-ssh', 'description': 'Allows TCP connections from any source to any instance on the network using port 22.', 'network': 'https://www.googleapis.com/compute/v1/projects/skypilot-managed-spot/global/networks/skypilot-vpc', 'priority': 65534, 'sourceRanges': ['0.0.0.0/0'], 'allowed': [{'IPProtocol': 'tcp', 'ports': ['22']}], 'direction': 'INGRESS', 'logConfig': {'enable': False}, 'disabled': False, 'selfLink': 'https://www.googleapis.com/compute/v1/projects/skypilot-managed-spot/global/firewalls/skypilot-vpc-allow-ssh'}]
```

Reply:
Thanks for the explanation! It would be nice to include the explanation in the code.

Reply:
added!